### 1) Случайная непрерывная величина A имеет равномерное распределение на промежутке (200, 800].
### Найдите ее среднее значение и дисперсию.

Используем формулы для определения мат. ожидания и дисперсии для случайной непрерывной величины X, равномерно распределённой на отрезке [a;b]:
Математическое ожидание: M(X)=(a+b)/2
Дисперсия: D(X)=〖(b-a)^2/12 
Вычислим:

In [1]:
def mean_and_variance(a,b):
    return f'На промежутке ({a};{b}]\nСреднее значение: М(А) = {(a+b)/2: .2f}\nДисперсия: D(A) = {((b-a)**2)/12: .2f}'
print(mean_and_variance(200,800))

На промежутке (200;800]
Среднее значение: М(А) =  500.00
Дисперсия: D(A) =  30000.00


#### Ответ:
**M(A) = 500.00**

**D(A) = 30000.00**

### 2) О случайной непрерывной равномерно распределенной величине B известно, что ее дисперсия равна 0.2.
### Можно ли найти правую границу величины B и ее среднее значение зная, что левая граница равна 0.5? Если да, найдите ее.

Зная, что дисперсия случайной непрерывной равномерно распределённой на отрезке (a;b)величины B находится по формуле:

D(X)=〖(b-a)〗^2/12

И имея заданные значения D(B)=0.2 и a=0.5

Найдем правую границу b величины B из уравнения

〖(b-0.5)〗^2/12=0.2b-0.5=√2.4 b=0.5+√2.4

И найдем среднее значение по формуле:

M(B)=(a+b)/2

In [3]:
b=0.5+2.4**(1/2)
print(f'Левая граница распределения величины В - b = {b: .3f}\n'
      f'Среднее значение В на промежутке (0.5; {b: .3f}) M(B) = {(b+0.5)/2: .3f}')

Левая граница распределения величины В - b =  2.049
Среднее значение В на промежутке (0.5;  2.049) M(B) =  1.275


#### Ответ:

**Левая граница распределения: b = 2.049**
    
**Среднее значение: M(B) = 1.275**


### 3) Непрерывная случайная величина X распределена нормально и задана плотностью распределения

### f(x) = (1 / (4 * sqrt(2pi))) * exp((-(x+2)**2) / 32)

### Найдите:
### а) M(X)     б) D(X)     в) std(X) (среднее квадратичное отклонение)

![решение](Задача_4_2.jpg)

#### Ответ:
**M(X) = a = -2; σ = 4; D(X) = σ^2= 16**

### 4) Рост взрослого населения города X имеет нормальное распределение, причем, средний рост равен 174 см, а среднее квадратическое отклонение равно 8 см. посчитайте, какова вероятность того, что случайным образом выбранный взрослый человек имеет рост:
**а. больше 182 см?**

**б. больше 190 см?**

**в. от 166 см до 190 см?**

**г. от 166 см до 182 см?**

**д. от 158 см до 190 см?**

**е. не выше 150 см или не ниже 190 см?**

**ё. не выше 150 см или не ниже 198 см?**

**ж. ниже 166 см?**

_Задачу можно решить двумя способами: без использования сторонних библиотек (numpy, scipy, pandas и пр.), а затем проверить себя с помощью встроенных функций_

Напишем функцию для определения Z-значений для данного распределения:

In [5]:
def z_value(height):
    return (height-174)/8

In [6]:
from statistics import NormalDist

a) Определим Z-значениe:

In [7]:
print(z_value(182))

1.0


метод cdf определяет вероятность того, что случайная величина X будет меньше или равна x, т.е в данном случае P(X ≤ 182)

In [8]:
NormalDist().cdf(z_value(182))

0.8413447460685428

Однако нам нужна вероятность P(X ≥ 182), поэтому вычтем полученное значение из 1:

In [9]:
P=1-NormalDist().cdf(z_value(182))
P

0.15865525393145719

_**P(X ≥ 182) = 0.1587**_

б) Определим Z-значениe:

In [10]:
z=z_value(190)
z

2.0

Определим **P(X ≥ 190)**

In [11]:
P=1-NormalDist().cdf(z)
P

0.022750131948179098

_**P(X ≥ 190) = 0.02275**_

в) Определим Z1 и Z2:

In [12]:
z1=z_value(166)
z1

-1.0

In [13]:
z2=z_value(190)
z2

2.0

Определим **P(166 ≤ X ≤ 190)**

In [14]:
P=NormalDist().cdf(z2)-NormalDist().cdf(z1)
P

0.8185946141203637

_**P(166 ≤ X ≤ 190) = 0.8186**_

г) Определим Z1 и Z2:

In [18]:
z1=z_value(166)
z1

-1.0

In [16]:
z2=z_value(182)
z2

1.0

Определим **P(166 ≤ X ≤ 182)**

In [17]:
P=NormalDist().cdf(z2)-NormalDist().cdf(z1)
P

0.6826894921370856

_**P(166 ≤ X ≤ 182) = 0.6827**_

д) Определим Z1 и Z2:

In [19]:
z1=z_value(158)
z1

-2.0

In [20]:
z2=z_value(190)
z2

2.0

Определим **P(158 ≤ X ≤ 190)**

In [22]:
P=NormalDist().cdf(z2)-NormalDist().cdf(z1)
P

0.9544997361036418

_**P(158 ≤ X ≤ 190) = 0.9545**_

е) Определим Z1 и Z2:

In [24]:
z1=z_value(150)
z1

-3.0

In [25]:
z2=z_value(190)
z2

2.0

Определим **P(X ≤ 150 ∪ X ≥ 190)**

In [26]:
P=NormalDist().cdf(z1)+(1-NormalDist().cdf(z2))
P

0.0241000299798092

_**P(X ≤ 150 ∪ X ≥ 190) = 0.0241**_

ё) Определим Z1 и Z2:

In [27]:
z1=z_value(150)
z1

-3.0

In [28]:
z2=z_value(198)
z2

3.0

Определим **P(X ≤ 150 ∪ X ≥ 198)**

In [29]:
P=NormalDist().cdf(z1)+(1-NormalDist().cdf(z2))
P

0.002699796063260207

_**P(X ≤ 150 ∪ X ≥ 190) = 0.0027**_

ж) Определим Z-значениe:

In [31]:
z=z_value(166)
z

-1.0

Определим **P(X ≤ 166)**

In [33]:
P=NormalDist().cdf(z)
P

0.15865525393145713

_**P(X ≤ 166) = 0.1587**_

#### Ответ:

_**а) P(X ≥ 182) = 0.1587**_

_**б) P(X ≥ 190) = 0.02275**_

_**в) P(166 ≤ X ≤ 190) = 0.8186**_

_**г) P(166 ≤ X ≤ 182) = 0.6827**_

_**д) P(158 ≤ X ≤ 190) = 0.9545**_

_**е) P(X ≤ 150 ∪ X ≥ 190) = 0.0241**_

_**ё) P(X ≤ 150 ∪ X ≥ 190) = 0.0027**_

_**ж) P(X ≤ 166) = 0.1587**_

### 5) На сколько сигм (средних квадратичных отклонений) отклоняется рост человека, равный 190 см, от математического ожидания роста в популяции, в которой M(X) = 178 см и D(X) = 25 кв.см?

Воспользуемся формулой расчета Z-оценки:
![решение](Задача_4_5.jpg)

In [36]:
Z=(190-178)/25**(1/2)
Z

2.4

In [37]:
print(f'Рост человека, равный 190 см, отклоняется от математического ожидания роста в популяции, \n'
      f'в которой M(X) = 178 см и D(X) = 25 кв.см на Z = {Z} сигм.')

Рост человека, равный 190 см, отклоняется от математического ожидания роста в популяции, 
в которой M(X) = 178 см и D(X) = 25 кв.см на Z = 2.4 сигм.


#### Ответ:
    
**на 2.4 сигмы**